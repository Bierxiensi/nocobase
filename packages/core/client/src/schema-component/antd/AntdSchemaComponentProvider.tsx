import * as components from '.';
import { Plugin } from '../../application-v2/Plugin';
import * as common from '../common';
import { useFilterActionProps } from './filter/useFilterActionProps';
import { requestChartData } from './g2plot/requestChartData';

export class AntdSchemaComponentPlugin extends Plugin {
  async load() {
    this.addComponents();
    this.addScopes();
  }

  addComponents() {
    this.app.addComponents({
      ...(components as any),
      ...common,
    });
  }

  addScopes() {
    this.app.addScopes({
      requestChartData,
      useFilterActionProps,
    });
  }
}
