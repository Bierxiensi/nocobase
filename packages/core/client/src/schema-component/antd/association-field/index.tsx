import { AssociationField } from './AssociationField';
import { AssociationFieldAddNewer } from './AssociationField.AddNewer';
import { AssociationFieldSelector } from './AssociationField.Selector';
import { AssociationFieldViewer } from './AssociationField.Viewer';
import { AssociationFieldNesterDecorator } from './AssociationField.Nester.Decorator';
import { Nester } from './AssociationField.Nester';
AssociationField.AddNewer = AssociationFieldAddNewer;
AssociationField.Selector = AssociationFieldSelector;
AssociationField.Viewer = AssociationFieldViewer;
AssociationField.Nester = Nester;
AssociationField.Nester.Decorator = AssociationFieldNesterDecorator;

export * from './components/SubForm';
export * from './components/TableField';

export { AssociationField };
