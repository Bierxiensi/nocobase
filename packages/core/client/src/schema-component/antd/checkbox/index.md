---
group:
  title: Schema Components
  order: 3
---

# Checkbox

## Examples

### 勾选

<code src="./demos/demo1.tsx"></code>

### 组

<code src="./demos/demo2.tsx"></code>
