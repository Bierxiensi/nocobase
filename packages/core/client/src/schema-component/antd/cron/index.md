---
nav:
  path: /client
group:
  path: /schema-components
---

# Cron

## Example

#### Cron

<code src="./demos/demo1.tsx"></code>

#### CronSet

<code src="./demos/demo2.tsx"></code>
