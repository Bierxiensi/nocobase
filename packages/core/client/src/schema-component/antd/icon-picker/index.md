---
group:
  title: Schema Components
  order: 3
---

# IconPicker

## Examples

### IconPicker

<code src="./demos/demo1.tsx"></code>
