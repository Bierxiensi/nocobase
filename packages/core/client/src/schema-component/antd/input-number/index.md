---
group:
  title: Schema Components
  order: 3
---

# InputNumber

## Examples

### InputNumber

<code src="./demos/demo1.tsx"></code>

### addonBefore/addonAfter

<code src="./demos/demo2.tsx"></code>

### High precision decimals

<code src="./demos/demo3.tsx"></code>
