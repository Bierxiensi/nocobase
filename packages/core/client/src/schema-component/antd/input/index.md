---
group:
  title: Schema Components
  order: 3
---

# Input

## Nodes

- Input
- Input.TextArea
- Input.URL

## Examples

### Input

<code src="./demos/demo1.tsx"></code>

### Textarea

<code src="./demos/demo2.tsx"></code>

### URL

<code src="./demos/demo3.tsx"></code>

### JSON

<code src="./demos/demo4.tsx"></code>
