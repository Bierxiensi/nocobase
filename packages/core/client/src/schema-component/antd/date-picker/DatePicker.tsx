import { connect, mapProps, mapReadPretty } from '@formily/react';
import { DatePicker as AntdDatePicker } from 'antd';
import type {
  DatePickerProps as AntdDatePickerProps,
  RangePickerProps as AntdRangePickerProps
} from 'antd/lib/date-picker';
import React from 'react';
import { ReadPretty } from './ReadPretty';
import { mapDatePicker, mapRangePicker } from './util';

type ComposedDatePicker = React.FC<AntdDatePickerProps> & {
  RangePicker?: React.FC<AntdRangePickerProps>;
};

export const DatePicker: ComposedDatePicker = connect(
  AntdDatePicker,
  mapProps(mapDatePicker()),
  mapReadPretty(ReadPretty.DatePicker),
);

DatePicker.RangePicker = connect(
  AntdDatePicker.RangePicker,
  mapProps(mapRangePicker()),
  mapReadPretty(ReadPretty.DateRangePicker),
);

export default DatePicker;
