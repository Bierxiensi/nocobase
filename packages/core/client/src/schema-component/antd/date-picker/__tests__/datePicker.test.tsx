import { FormItem } from '@formily/antd';
import React from 'react';
import { render } from '@testing-library/react';
import { SchemaComponent, SchemaComponentProvider } from '../../../../schema-component';
import { Input } from '../../input';
import { DatePicker } from '../DatePicker';

const schema = {
  type: 'object',
  properties: {
    input: {
      type: 'boolean',
      title: `Editable`,
      'x-decorator': 'FormItem',
      'x-component': 'DatePicker',
      'x-component-props': {
        dateFormat: 'YYYY/MM/DD',
        showTime: true,
      },
      // default: '2022-11-22',
    },
    read1: {
      type: 'boolean',
      title: `Read pretty`,
      'x-read-pretty': true,
      'x-decorator': 'FormItem',
      'x-component': 'DatePicker',
      'x-component-props': {
        dateFormat: 'YYYY/MM/DD',
        showTime: true,
      },
    },
  },
};

const App = () => {
  return (
    <SchemaComponentProvider components={{ Input, DatePicker, FormItem }}>
      <SchemaComponent schema={schema} />
    </SchemaComponentProvider>
  );
};

describe('DatePicker', () => {
  it('DatePicker', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
