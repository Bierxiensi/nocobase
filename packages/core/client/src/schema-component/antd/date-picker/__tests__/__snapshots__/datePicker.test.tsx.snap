// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker DatePicker 1`] = `
<div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            Editable
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div
            class="ant-picker"
          >
            <div
              class="ant-picker-input"
            >
              <input
                autocomplete="off"
                placeholder="Select date"
                readonly=""
                size="21"
                title=""
                value=""
              />
              <span
                class="ant-picker-suffix"
              >
                <span
                  aria-label="calendar"
                  class="anticon anticon-calendar"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="calendar"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                    />
                  </svg>
                </span>
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="ant-formily-item ant-formily-item-layout-horizontal ant-formily-item-feedback-layout-loose ant-formily-item-label-align-right ant-formily-item-control-align-left"
  >
    <div
      class="ant-formily-item-label"
    >
      <div
        class="ant-formily-item-label-content"
      >
        <span>
          <label>
            Read pretty
          </label>
        </span>
      </div>
      <span
        class="ant-formily-item-colon"
      >
        :
      </span>
    </div>
    <div
      class="ant-formily-item-control"
    >
      <div
        class="ant-formily-item-control-content"
      >
        <div
          class="ant-formily-item-control-content-component"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
